src/backend/core/models.py

```diff
@@ -325,3 +387,225 @@ def clean_fields(self, exclude=None):
+            "update": False,
+            "partial_update": False,
+            "retrieve": False,
+            "set_role_to": [],
+        }
+
+        # Early returns for unauthenticated users
+        if not user.is_authenticated:
+            return abilities
+
+        roles = get_resource_roles(resource, user)
+
+        # if no roles on the resource early return with no abilities
+        if not roles:
+            return abilities
+
+        abilities["retrieve"] = True
+
+        is_admin = RoleChoices.ADMIN in roles
+        is_owner = RoleChoices.OWNER in roles
+
+        # Restrict to read-only if insufficient privileges
+        if not (is_admin or is_owner) or (
+            self.role == RoleChoices.OWNER and not is_owner
+        ):
+            return abilities
+
+        # Allow owner deletion only if not the last owner
+        if self.role == RoleChoices.OWNER:
+            if is_owner:
+                owner_count = resource.accesses.filter(role=RoleChoices.OWNER).count()
+                abilities["destroy"] = owner_count > 1
+        else:
+            abilities["destroy"] = is_owner or is_admin
+
+        # Determine role modification permissions
+        can_modify_roles = abilities["destroy"] and not (
+            self.role == RoleChoices.OWNER and not is_owner
+        )
+        assignable_roles = self._get_assignable_roles(can_modify_roles, is_owner)
+
+        abilities.update(
+            {
+                "update": bool(assignable_roles),
+                "partial_update": bool(assignable_roles),
+                "set_role_to": assignable_roles,
+            }
+        )
+
+        return abilities
+
+
+class Recording(BaseModel):
+    """Model for recordings that take place in a room"""
+
+    room = models.ForeignKey(
+        Room,
+        on_delete=models.CASCADE,
+        related_name="recordings",
+        verbose_name=_("Room"),
+    )
+    status = models.CharField(
+        max_length=20,
+        choices=RecordingStatusChoices.choices,
+        default=RecordingStatusChoices.INITIATED,
+    )
+    worker_id = models.CharField(
+        max_length=255,
+        null=True,
+        blank=True,
+        verbose_name=_("Worker ID"),
+        help_text=_(
+            "Enter an identifier for the worker recording."
+            "This ID is retained even when the worker stops, allowing for easy tracking."
+        ),
+    )
+
+    class Meta:
+        db_table = "meet_recording"
+        ordering = ("-created_at",)
+        verbose_name = _("Recording")
+        verbose_name_plural = _("Recordings")
+        constraints = [
+            models.UniqueConstraint(
+                fields=["room"],
+                condition=models.Q(status=RecordingStatusChoices.ACTIVE)
+                | models.Q(status=RecordingStatusChoices.INITIATED),
+                name="unique_initiated_or_active_recording_per_room",
+            )
+        ]
+
+    def __str__(self):
+        return str(self.id)
```

Comment:
Be careful, because you might end up with an UUID printed/logged without knowing what object it is. I would rather do
```suggestion
        return f"Recording {self.id} (self.status)"  # status is not really needed
```

Reply:
I was using it for the filename, but you're true.
I will create a proper getter method with an explicit naming.